scripts/irc.js

```
@@ -0,0 +1,196 @@
+
+
+/*----------  Function Declaration  ----------*/
+
+const addChatMessage = (author, newMessage) => {
+    let messageToInsert = document.createElement('li');
+    messageToInsert.className = 'irc-modal__chat-message';
+    let messageAuthor = document.createElement('span');
+    messageAuthor.className = 'irc-modal__chat-author';
+    messageAuthor.innerHTML = author + ' ';
+    let messageText = document.createTextNode(newMessage);
+    messageToInsert.appendChild(messageAuthor);
+    messageToInsert.appendChild(messageText);
+
+    ircChatLog.appendChild(messageToInsert);
+}
+
+const greetUser = (event, userMessage) => {
+    const punctuationRegex = /[\u2000-\u206F\u2E00-\u2E7F\\'!"#$%&()*+,\-.\/:;<=>?@\[\]^_`{|}~]/;
+    const spaceRegex = /\s+/g;
+
+    switch (userMessage.toLowerCase().replace(punctuationRegex, '').replace(spaceRegex, ' ')) {
+        case 'hi':
+        case 'hello':
+        case 'good morning':
+        case 'good afternoon':
+        case 'gm':
+        case 'yo':
+        case 'sup':
+        case 'heya':
+        case 'hey':
+        case 'howdy':
+            let greeterNumber = Math.floor(Math.random() * memberNameList.length);
+            console.log("time to greet!");
+            for (let i = 0; i < greeterNumber; i++) {
+                setTimeout(() => {
+                    let author = selectAtRandom(memberNameList);
+                    if (author !== 'windauxs9x') {
+                        addChatMessage(author, selectAtRandom(greetings));
+                    }
+
+                }, 2000);
+            }
+            break;
+    }
+}
+
+const selectAtRandom = (arrayToSelectFrom) => {
+    if (arrayToSelectFrom !== undefined || arrayToSelectFrom.length !== 0) {
+        return arrayToSelectFrom[Math.floor(Math.random() * arrayToSelectFrom.length)];
+    }
+}
```

consider throwing an error with guard clauses rather than only return value if these cases are met

---

```js
 const randomElementFromArray = (arr) => {
 if(arr.length < 1) {
 throw new Error()
 } 
 if(!Array.isArray(arr)) {
  throw new Error()
 } 
 const randomIndex = Math.floor(Math.random() *arr.length) 
 return arr[randomIndex]
 } 
```